forc/src/cli/commands/test.rs

```diff
@@ -90,5 +90,14 @@ pub(crate) fn exec(command: Command) -> Result<()> {
     out.lines().for_each(|line| println!("{}", line.unwrap()));
     thread.join().unwrap();
 
-    Ok(())
+    let child_success = match child.try_wait() {
+        Ok(Some(returned_status)) => returned_status.success(),
+        Ok(None) => child.wait().unwrap().success(),
+        Err(_) => false,
+    };
+
+    match child_success {
+        true => Ok(()),
+        false => bail!("child test process failed"),
```

Comment:
> You could also simplify this to just `bail!("child test process failed")`

I continued with `bail!` since this comment from @eureka-cpu got a 👍 from @JoshuaBatty (and it was more recent) but wasn't sure whether it is preferred over `anyhow!`. Maybe I can request a re-review from @JoshuaBatty

Reply:
Re-reviews should be requested anyways, otherwise the reviewer doesn't know when they should review again!

Reply:
@adlerjohn `bail!` is essentially just a shorthand macro from anyhow for `return Err(anyhow!(...`